contracts/managers/allocation-pricers/BinaryAllocationPricer.sol

```diff
@@ -126,75 +126,169 @@ contract BinaryAllocationPricer is
         // Set Core and setTokenFactory
         coreInstance = _coreInstance;
         setTokenFactoryAddress = _setTokenFactoryAddress;
+
+        // Store passed in collateral in mapping
+        bytes32 baseCollateralHash = calculateCollateralIDHashFromSet(
+            _baseAssetCollateralInstance
+        );
+        bytes32 quoteCollateralHash = calculateCollateralIDHashFromSet(
+            _quoteAssetCollateralInstance
+        );
+        storedCollateral[baseCollateralHash] = address(_baseAssetCollateralInstance);
+        storedCollateral[quoteCollateralHash] = address(_quoteAssetCollateralInstance);
+
+        emit NewCollateralLogged(baseCollateralHash, address(_baseAssetCollateralInstance));
+        emit NewCollateralLogged(quoteCollateralHash, address(_quoteAssetCollateralInstance));
     }
 
+    /* ============ External ============ */
+
     /*
      * Determine the next allocation to rebalance into. If the dollar value of the two collateral sets is more
```

Comment:
What does `newAllocation` mean? It seems this function is not only figuring out which Set is next, but also returning the valuations of current and next. Is there a way to separate these functions?